packages/stream-tests/src/__tests__/basic-indexing.test.ts

```diff
@@ -0,0 +1,253 @@
+    expect(results[0].content).toEqual(CONTENT1)
+    expect(results[1].id.toString()).toEqual(doc2.id.toString())
+    expect(results[1].content).toEqual(CONTENT2)
+    expect(results[2].id.toString()).toEqual(doc3.id.toString())
+    expect(results[2].content).toEqual(CONTENT3)
+    expect(results[3].id.toString()).toEqual(doc4.id.toString())
+    expect(results[3].content).toEqual(CONTENT4)
+    expect(results[4].id.toString()).toEqual(doc5.id.toString())
+    expect(results[4].content).toEqual(CONTENT5)
+  })
+
+  test('multiple documents - one page, backwards iteration', async () => {
+    const doc1 = await ModelInstanceDocument.create(ceramic, CONTENT0, midMetadata)
+    await doc1.replace(CONTENT1)
+    const doc2 = await ModelInstanceDocument.create(ceramic, CONTENT2, midMetadata)
+    const doc3 = await ModelInstanceDocument.create(ceramic, CONTENT3, midMetadata)
+
+    console.log(`docIds: [${doc1.id.toString()}, ${doc2.id.toString()}, ${doc3.id.toString()}]`)
+
+    const resultObj = await ceramic.index.queryIndex({ model: model.id, last: 100 })
+    const results = await extractDocuments(ceramic, resultObj)
+
+    // Using `last` doesn't change the order of documents returned within each page
+    expect(results.length).toEqual(3)
+    expect(results[0].id.toString()).toEqual(doc1.id.toString())
+    expect(results[0].content).toEqual(CONTENT1)
+    expect(results[1].id.toString()).toEqual(doc2.id.toString())
+    expect(results[1].content).toEqual(CONTENT2)
+    expect(results[2].id.toString()).toEqual(doc3.id.toString())
+    expect(results[2].content).toEqual(CONTENT3)
+  })
+
+  test('multiple documents - multiple pages, backwards iteration', async () => {
+    const doc1 = await ModelInstanceDocument.create(ceramic, CONTENT0, midMetadata)
+    await doc1.replace(CONTENT1)
+    const doc2 = await ModelInstanceDocument.create(ceramic, CONTENT2, midMetadata)
+    const doc3 = await ModelInstanceDocument.create(ceramic, CONTENT3, midMetadata)
+    const doc4 = await ModelInstanceDocument.create(ceramic, CONTENT4, midMetadata)
+    const doc5 = await ModelInstanceDocument.create(ceramic, CONTENT5, midMetadata)
+
+    const resultObj0 = await ceramic.index.queryIndex({ model: model.id, last: 2 })
+    expect(resultObj0.pageInfo.hasPreviousPage).toBeTruthy()
+    const resultObj1 = await ceramic.index.queryIndex({
+      model: model.id,
+      last: 2,
+      before: resultObj0.pageInfo.startCursor,
+    })
+    expect(resultObj1.pageInfo.hasPreviousPage).toBeTruthy()
+    const resultObj2 = await ceramic.index.queryIndex({
+      model: model.id,
+      last: 2,
+      before: resultObj1.pageInfo.startCursor,
+    })
+    expect(resultObj2.pageInfo.hasPreviousPage).toBeFalsy()
+
+    const results = (
+      await Promise.all([
+        extractDocuments(ceramic, resultObj0),
+        extractDocuments(ceramic, resultObj1),
+        extractDocuments(ceramic, resultObj2),
```

Comment:
I think this is where the test doesn't match how GraphQL clients handle the results, if instead it was
```suggestion
        extractDocuments(ceramic, resultObj2),
        extractDocuments(ceramic, resultObj1),
        extractDocuments(ceramic, resultObj0),
```
then the results order would be consistent.